### ISSUE TITLE
Comptime generated wrapper causes many issues

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1352]
I wrote a wrapper generator so I could write a drop-in "shared-ptr", but this causes many issues... Here are some that I found:

- The IDE gets really confused, things like squiggles appearing for some time and then disappearing is quite common.
- Hot compilation is completely broke.
- If you write something like `Class.StaticMember.Identifier` the IDE crashes. (example: `IDEApp.sApp.AddNewProjectToWorkspace`)

If you go to the IDE and open the file IDEApp.bf at line 466 and replace it with `public Wrapper<List<ExecutionCmd>> mExecutionQueue = .(new List<ExecutionCmd>()) ~ DeleteContainerAndItems!(_.Obj);`, you will be able to reproduce all these bugs and even more, since looks like the IDE gives up trying to do anything and hangs if you try to invoke auto-complete, build or clean.

<details>
  <summary>Code for Wrapper / Wrapper Generator</summary>

```cs
using System;

namespace System
{
	[GenerateWrapper]
	public struct Wrapper<T>
		where T : delete
	{
		private T mObj;

		public T Obj => mObj;

		public this(T obj)
		{
			mObj = obj;
		}
	}
}
```

```cs
using System;
using System.Collections;
using System.Reflection;
using System.Diagnostics;

namespace System
{
	[AttributeUsage(.Struct | .Class)]
	struct GenerateWrapperAttribute : Attribute, IComptimeTypeApply
	{
		private enum PropertyMethods
		{
			Undefined = 0,
			Get = 1,
			Set = 2,
			GetSet = Get | Set
		}

		private class PropertyBase
		{
			public Type Type;
			public PropertyMethods Methods;
		}

		private class NormalProperty : PropertyBase
		{
		}

		private class IndexerProperty : PropertyBase
		{
			public List<(StringView, Type)> Parameters = new .() ~ delete _;
		}

		[Comptime]
		private Type UnrefType(Type type)
		{
			if (let refType = type as RefType)
				return refType.UnderlyingType;
			return type;
		}

		[Comptime]
		public void ApplyToType(Type type)
		{
			let specializedType = type as SpecializedGenericType;
			if (specializedType == null)
				return;

			let genArgType = specializedType.GetGenericArg(0);

			for (let iType in genArgType.Interfaces)
			{
				// hack to ignore IList from List<T>, since idk how to implement it's method
				if (iType == typeof(IList))
					continue;
				Compiler.EmitAddInterface(type, iType);
			}


			String code = scope .();

			Dictionary<StringView, PropertyBase> properties = scope .();

			for (let methodInfo in genArgType.GetMethods())
			{
				if (methodInfo.IsConstructor || methodInfo.IsDestructor)
					continue;

				if (methodInfo.Name == "~this" || methodInfo.Name == "this" || methodInfo.Name.StartsWith("this$"))
					continue;

				// hack to ignore private members from List<T>
				if (methodInfo.Name == "GCMarkMembers" || methodInfo.ReturnType == typeof(Variant))
					continue;

				if (methodInfo.Name == "get__")
				{
					let propType = UnrefType(methodInfo.ReturnType);
					let propertyName = ((int)propType.TypeId).ToString(.. scope .());
					if (properties.TryAdd(propertyName, let keyPtr, let valuePtr))
					{
						let indexerProp = scope:: IndexerProperty();
						indexerProp.Type = propType;
						indexerProp.Methods = .Get;
						for (int i = 0; i < methodInfo.ParamCount; i++)
							indexerProp.Parameters.Add((methodInfo.GetParamName(i), methodInfo.GetParamType(i)));
						*valuePtr = indexerProp;
					}
					else
					{
						IndexerProperty indexerProp = (.)*valuePtr;
						Debug.Assert(indexerProp.Type == propType);
						if (!indexerProp.Methods.HasFlag(.Get))
							indexerProp.Methods |= .Get;
					}
				}
				else if(methodInfo.Name == "set__")
				{
					let propertyName = ((int)methodInfo.GetParamType(0).TypeId).ToString(.. scope .());
					if (properties.TryAdd(propertyName, let keyPtr, let valuePtr))
					{
						let indexerProp = scope:: IndexerProperty();
						indexerProp.Type = methodInfo.GetParamType(0);
						indexerProp.Methods = .Set;
						for (int i = 1; i < methodInfo.ParamCount; i++)
							indexerProp.Parameters.Add((methodInfo.GetParamName(i), methodInfo.GetParamType(i)));
						*valuePtr = indexerProp;
					}
					else
					{
						IndexerProperty indexerProp = (.)*valuePtr;
						Debug.Assert(indexerProp.Type == methodInfo.GetParamType(0));
						if (!indexerProp.Methods.HasFlag(.Set))
							indexerProp.Methods |= .Set;
					}
				}
				else if (methodInfo.Name.StartsWith("get__"))
				{
					let propertyName = methodInfo.Name..RemoveFromStart(5);
					if (properties.TryAdd(propertyName, let keyPtr, let valuePtr))
					{
						let prop = scope:: NormalProperty();
						prop.Type = methodInfo.ReturnType;
						prop.Methods = .Get;
						*valuePtr = prop;
					}
					else
					{
						NormalProperty prop = (.)*valuePtr;
						Debug.Assert(prop.Type == methodInfo.ReturnType);
						if (!prop.Methods.HasFlag(.Get))
							prop.Methods |= .Get;
					}
				}
				else if (methodInfo.Name.StartsWith("set__"))
				{
					let propertyName = methodInfo.Name..RemoveFromStart(5);
					if (properties.TryAdd(propertyName, let keyPtr, let valuePtr))
					{
						let prop = scope:: NormalProperty();
						prop.Type = methodInfo.GetParamType(0);
						prop.Methods = .Set;
						*valuePtr = prop;
					}
					else
					{
						NormalProperty prop = (.)*valuePtr;
						Debug.Assert(prop.Type == methodInfo.GetParamType(0));
						if (!prop.Methods.HasFlag(.Set))
							prop.Methods |= .Set;
					}
				}
				else if (methodInfo.Name == "")
				{
					Debug.Assert(methodInfo.ParamCount == 1);

					code.AppendF($"""
						[Inline] public static operator comptype(Type.[Friend]GetType((.){(int)methodInfo.ReturnType.TypeId}))(
						""");

					//for (int i < methodInfo.ParamCount)
					//{
					let paramName = methodInfo.GetParamName(0);
					//	let paramType = methodInfo.GetParamType(0);

					code.AppendF($"""
						Self {paramName}
						""");

					//	if (i != methodInfo.ParamCount - 1)
					//		code.Append(", ");
					//}

					code.AppendF($") => {paramName}.mValue;\n");
				}
				else
				{
					code.AppendF($"""
						public comptype(Type.[Friend]GetType((.){(int)methodInfo.ReturnType.TypeId})) {methodInfo.Name}(
						""");

					for (int i < methodInfo.ParamCount)
					{
						let paramName = methodInfo.GetParamName(i);
						let paramType = methodInfo.GetParamType(i);

						code.AppendF($"""
							comptype(Type.[Friend]GetType((.){(int)paramType.TypeId})) {paramName}
							""");

						if (i != methodInfo.ParamCount - 1)
							code.Append(", ");
					}

					code.AppendF($") => mObj.{methodInfo.Name}(");

					for (int i < methodInfo.ParamCount)
					{
						let paramName = methodInfo.GetParamName(i);

						code.AppendF($"{paramName}");

						if (i != methodInfo.ParamCount - 1)
							code.Append(", ");
					}

					code.AppendF(");\n");
				}
			}

			for (let (propertyName, propertyInfo) in properties)
			{
				if (let indexerProp = propertyInfo as IndexerProperty)
				{
					code.AppendF($"public comptype(Type.[Friend]GetType((.){(int)propertyInfo.Type.TypeId})) this[");

					for (let (paramName, paramType) in indexerProp.Parameters)
					{
						code.AppendF($"""
							comptype(Type.[Friend]GetType((.){(int)paramType.TypeId})) {paramName}
							""");

						if (@paramName.Index != indexerProp.Parameters.Count - 1)
							code.Append(", ");
					}

					code.AppendF($"]\n{{\n");

					if (indexerProp.Methods.HasFlag(.Get))
					{
						code.Append("\t[Inline] get => mObj[");

						for (let (paramName, paramType) in indexerProp.Parameters)
						{
							code.AppendF($"{paramName}");

							if (@paramName.Index != indexerProp.Parameters.Count - 1)
								code.Append(", ");
						}

						code.Append("];\n");
					}

					if (indexerProp.Methods.HasFlag(.Set))
					{
						code.Append("\t[Inline] set => mObj[");

						for (let (paramName, paramType) in indexerProp.Parameters)
						{
							code.AppendF($"{paramName}");

							if (@paramName.Index != indexerProp.Parameters.Count - 1)
								code.Append(", ");
						}

						code.Append("] = value;\n");
					}

					code.Append("}\n");
				}
				else if (let normalProp = propertyInfo as NormalProperty)
				{
					code.AppendF($"public comptype(Type.[Friend]GetType((.){(int)propertyInfo.Type.TypeId})) {propertyName}\n{{\n");

					if (normalProp.Methods.HasFlag(.Get))
						code.AppendF($"\t[Inline] get => mObj.{propertyName};\n");

					if (normalProp.Methods.HasFlag(.Set))
						code.AppendF($"\t[Inline] set => mObj.{propertyName} = value;\n");

					code.Append("}\n");
				}
				else
					Runtime.FatalError("Unknown property type");
			}

			Compiler.EmitTypeBody(type, code);
		}
	}
}
```
</details>

**Comment by bfiete** [comment]
Hm, also - maybe it would be convenient if `comptype` could also just take an int typeid...

**Comment by bfiete** [comment]
I believe these are all fixed here 0c3f9a139d88c6a52141440df1b4e4a60fb3150a
Also you can simplify  `comptype(Type.[Friend]GetType((.){(int)paramType.TypeId}))` to `comptype({(int)paramType.TypeId})` now
a65b69bcd008d767532cc73cf05f486894c276e9
